ci/jenkins/test-vm.sh

```diff
@@ -0,0 +1,254 @@
+    export GOVC_USERNAME=$GOVC_USERNAME
+    export GOVC_PASSWORD=$GOVC_PASSWORD
+    export GOVC_INSECURE=1
+    export GOVC_DATACENTER=$GOVC_DATACENTER
+    export GOVC_DATASTORE=$GOVC_DATASTORE
+}
+
+function clean_up_one_ns {
+    ns=$1
+    kubectl get pod -n "${ns}" --no-headers=true | awk '{print $1}' | while read pod_name; do
+        kubectl delete pod "${pod_name}" -n "${ns}" --force --grace-period 0
+    done
+    kubectl delete ns "${ns}" --ignore-not-found=true || true
+}
+
+function clean_antrea {
+    echo "====== Cleanup Antrea Installation ======"
+    clean_up_one_ns $TEST_NAMESPACE
+    kubectl delete -f ${WORKDIR}/antrea.yml --ignore-not-found=true
+}
+
+function apply_antrea {
+    # Get the latest ubuntu image
+    make build-ubuntu
+    echo "====== Applying Antrea yaml ======"
+    ./hack/generate-manifest.sh --enable-nodeport --enable-externalnode > ${WORKDIR}/antrea.yml
+    kubectl apply -f ${WORKDIR}/antrea.yml
+    # TO BE DELETED
+    #kubectl get nodes -o wide --no-headers=true | awk -v role="$CONTROL_PLANE_NODE_ROLE" '$3 != role {print $6}' | while read IP; do
+    #    echo "Copying new antrea.yml to ${CONTROL_PLANE_NODE_ROLE} node ${IP} home directory"
+    #    scp -q -o StrictHostKeyChecking=no -i "${WORKDIR}/jenkins_id_rsa" ${WORKDIR}/antrea.yml ubuntu@${IP}:~
+    #done
+}
+
+function clean_vm_agent {
+    echo "====== Cleanup Antrea-Agent Installation on the VM ======"
+    echo "Revert to snapshot TestAgent"
+    govc snapshot.revert -k=true -vm=${LIN_HOSTNAME} TestAgent
+    govc snapshot.revert -k=true -vm=${WIN_HOSTNAME} TestAgent
+    echo "Get IP addresses for VMs"
+    UBUNTU_VM_IP=$(govc vm.ip -k=true -wait=1m ${LIN_HOSTNAME})
+    WINDOWS_VM_IP=$(govc vm.ip -k=true -wait=1m ${WIN_HOSTNAME})
+    echo "IP address of ubuntu VM ${UBUNTU_VM_IP} and windows VM ${WINDOWS_VM_IP}"
+    kubectl delete sa $SERVICE_ACCOUNT -n $TEST_NAMESPACE --ignore-not-found=true
+    clean_up_one_ns $TEST_NAMESPACE
+    echo "Deleting stale antrea-agent files from ${UBUNTU_VM_IP} and ${WINDOWS_VM_IP}"
+    ssh -o StrictHostKeyChecking=no -i "${WORKDIR}/jenkins_id_rsa" -n ubuntu@${UBUNTU_VM_IP} "rm -rf /tmp/antrea"
+    ssh -o StrictHostKeyChecking=no -i "${WORKDIR}/jenkins_id_rsa" Administrator@${WINDOWS_VM_IP} "rm -rf /tmp/antrea"
+}
+
+function configure_vm_agent {
+    echo "====== Configuring Antrea agent on the VM ======"
+    echo "Create ns $TEST_NAMESPACE"
+    kubectl create ns $TEST_NAMESPACE
+    echo "Create service account $SERVICE_ACCOUNT"
+    kubectl create sa $SERVICE_ACCOUNT -n $TEST_NAMESPACE
+    cp ./build/yamls/externalnode/vm-agent-rbac.yml ${WORKDIR}/vm-agent-rbac.yml
+    echo "Applying vm-agent rbac yaml"
+    kubectl apply -f ${WORKDIR}/vm-agent-rbac.yml
+
+    echo "Creating files antrea-agent.kubeconfig and antrea-agent.antrea.kubeconfig"
+    # Kubeconfig to access K8S API
+    APISERVER=$(kubectl config view -o jsonpath="{.clusters[?(@.name==\"$CLUSTER_NAME\")].cluster.server}")
```

Comment:
Is it easier if we provide a script to generate the kubeconfig files, and call it here directly? The script can be used by users too.